### PR TITLE
weechat: update to version 4.0.4, move to Python 3.10.

### PR DESCRIPTION
**Pull request by OscarL**
Lightly tested on beta4 64 bits (was able to connect and chat over the OFTC's `#haiku` channel).

### CODE REVIEW

File: net-irc/weechat/weechat-4.0.4.recipe [hunk 1]
```diff
@@ -54,7 +54,7 @@ BUILD_REQUIRES="
 	devel:libintl$secondaryArchSuffix
 	devel:liblua$secondaryArchSuffix >= 5.3
 	devel:libncurses$secondaryArchSuffix
-	devel:libpython3.9$secondaryArchSuffix
+	devel:libpython3.10$secondaryArchSuffix
```
Comment (augiedoggie) [inline net-irc/weechat/weechat-4.0.4.recipe:57]: Seems like it should also have a `lib:libpython3.10$secondaryArchSuffix` in the REQUIRES too.  Does it actually link against `libpython`?

Comment (augiedoggie) [reply net-irc/weechat/weechat-4.0.4.recipe:57]: I see it listed as an optional dependency for Arch Linux and I don't see libpython in the `ldd` output so I guess it's ok.  Haiku/Haikuporter could use a RECOMMENDS variable.

Comment (Begasus) [reply net-irc/weechat/weechat-4.0.4.recipe:57]: My guess is that it will use what is available at runtime, same as for Geany.

Comment (augiedoggie) [reply net-irc/weechat/weechat-4.0.4.recipe:57]: Right, but some of the optional dependencies are listed, like lua, but python isn't.

Comment (Begasus) [reply net-irc/weechat/weechat-4.0.4.recipe:57]: Maybe not in their docs, in Arch they are listed as optional: https://archlinux.org/packages/extra/x86_64/weechat/

**Pull request merged**